### PR TITLE
#1173 fixing automatic vacation

### PR DESCRIPTION
**Pull request by krzyk**
#1173 
* added a check if there are any positive awards for the last 16 days - this should prevent of automatic vacation
* added test to cover that

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#1181](https://github.com/zerocracy/farm/issues/1181) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#1181](https://github.com/zerocracy/farm/issues/1181) is assigned to @carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda), here is [why](https://www.0crat.com/footprint/C3NDPUA8L/1701048513); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

File: src/test/resources/com/zerocracy/bundles/dont_turn_on_vacation_for_too_many_negative_awards_if_there_are_positives/_before.groovy [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+import java.time.ZonedDateTime
+
+@SuppressWarnings('UnnecessaryObjectReferences')
+def exec(Project project, XML xml) {
+  Farm farm = binding.variables.farm
+  new ExtGithub(farm).value().repos()
+    .create(new Repos.RepoCreate('test', false))
+  Awards awards = new Awards(farm, 'krzyk').bootstrap()
+  ZonedDateTime now = ZonedDateTime.now()
+  awards.add(
```
Comment (carlosmiranda) [inline src/test/resources/com/zerocracy/bundles/dont_turn_on_vacation_for_too_many_negative_awards_if_there_are_positives/_before.groovy]: @krzyk we can make these tests a bit easier to read, and avoid `UnnecessaryObjectReferences` warning by using [with() method](http://mrhaki.blogspot.com/2009/09/groovy-goodness-with-method.html):

```groovy
awards.with {
  add project, 15, 'gh:test/test#1', 'test' new Date(now.minusDays(9).toInstant().toEpochMilli())
  add project, -15, 'gh:test/test#1', 'test', new Date(now.minusDays(8).toInstant().toEpochMilli())
  add project, -15, 'gh:test/test#1', 'test', new Date(now.minusDays(7).toInstant().toEpochMilli())
  // and so on
}
```

**Comment by krzyk** [comment]
@carlosmiranda updated, please check

**Comment by g4s8** [comment]
@rultor merge

**Comment by 0crat** [comment]
There is an unrecoverable failure on my side. Please, submit it [here](https://github.com/zerocracy/farm/issues):

```
PID: 4@2a5ea468-7e77-489c-80d0-c8be03a5d7e1, thread: AsyncFlush-4
com.zerocracy.farm.sync.SyncProject[77] java.lang.IllegalStateException: Failed to acquire "verbosity/carlosmiranda.xml" in "PMO" in 1min: 61df0a5983f1/8min/0/3/true/false by QuartzScheduler_Worker-1
```

[0.22.7](https://github.com/zerocracy/farm): CID: [1711062722](https://www.0crat.com/C3NDPUA8L/1711062722), Type: "Close job", Author: "rultor"

**Comment by 0crat** [comment]
There is an unrecoverable failure on my side. Please, submit it [here](https://github.com/zerocracy/farm/issues):

```
PID: 4@2a5ea468-7e77-489c-80d0-c8be03a5d7e1, thread: AsyncFlush-4
com.zerocracy.farm.sync.SyncProject[77] java.lang.IllegalStateException: Failed to acquire "verbosity/carlosmiranda.xml" in "PMO" in 1min: 61df0a5983f1/8min/0/3/true/false by QuartzScheduler_Worker-1
```

[0.22.7](https://github.com/zerocracy/farm): CID: [1711062722](https://www.0crat.com/C3NDPUA8L/1711062722), Type: "Close job", Author: "rultor"

**Comment by carlosmiranda** [comment]
@0crat status

**Comment by 0crat** [comment]
There is an unrecoverable failure on my side. Please, submit it [here](https://github.com/zerocracy/farm/issues):

```
PID: 4@a20fdf63-8643-4773-9ea4-95ef1507c623, thread: pool-57-thread-1
com.zerocracy.farm.sync.SyncProject[77] java.lang.IllegalStateException: Failed to acquire "people.xml" in "PMO" in 1min: 8c5662c87aa6/30min/0/3/true/false by AsyncFlush-2
```

[0.22.19](https://github.com/zerocracy/farm): Issue: zerocracy/farm#1181, Comment: 398914217

**Comment by carlosmiranda** [comment]
@0crat status

**Comment by 0crat** [comment]
> @0crat status ([here](https://github.com/zerocracy/farm/issues/1181#issuecomment-399050814))

@carlosmiranda This is what I know about this job in [C3NDPUA8L](https://www.0crat.com/p/C3NDPUA8L), as in [§32](http://www.zerocracy.com/policy.html#32):

  * The job [#1181](https://github.com/zerocracy/farm/issues/1181) is [in scope](http://www.zerocracy.com/policy.html#14) for 2days
  * The role is `REV`
  * The job is assigned to @carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda) for [2days](http://www.zerocracy.com/policy.html#8)
  * There is a monetary reward attached
  * @carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda) will get 25% of their hourly rate for this job
  * Some equity will be [vested](http://www.zerocracy.com/policy.html#37) on completion
  * The job doesn't have any [impediments](http://www.zerocracy.com/policy.html#9)
  * The [budget](http://www.zerocracy.com/policy.html#4) is 15 minutes/points
  * These users are banned and won't be assigned:
    * @krzyk[/z](https://www.0crat.com/u/krzyk): This user reported the ticket
  * The job is waiting quality review verdict by @ypshenychka[/z](https://www.0crat.com/u/ypshenychka) for 22hr
  * @carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda) will receive 15 minutes after quality review
  * Job [footprint](https://www.0crat.com/footprint/C3NDPUA8L?q=%7Bjob%3A%27gh%3Azerocracy%2Ffarm%231181%27%2Ctype%3A%7B%24not%3A%2F%5ENotify%2F%7D%7D) (restricted area)

**Comment by ypshenychka** [comment]
@carlosmiranda According to our [QA Rules](http://www.zerocracy.com/policy.html#42):
>The code reviewer found at least three problems in the code.
Comments were mostly about design problems, not cosmetic issues.

Only one issue was found during code review. 
Please confirm that you'll try to find at least three major problems while future reviews.

**Comment by carlosmiranda** [comment]
@ypshenychka confirmed

**Comment by ypshenychka** [comment]
@carlosmiranda Thanks

**Comment by ypshenychka** [comment]
@0crat quality acceptable

**Comment by ypshenychka** [comment]
@0crat ping

**Comment by 0crat** [comment]
> @0crat ping ([here](https://github.com/zerocracy/farm/issues/1181#issuecomment-399081392))

@ypshenychka Can't understand "ping", try one of these:
  - `assign` Assign a performer to this issue
  - `boost` Set a boost for the job
  - `hello` Just say hello
  - `in` Register this issue in scope (WBS)
  - `out` Close the order and remove this job from scope
  - `pay` Pay a user some extra cash
  - `quality` Review a task
  - `resign` Resign from this issue
  - `status` Check the status of the job
  - `version` Print current version of the bot
  - `wait` Register an impediment for a job

**Comment by 0crat** [comment]
Order was finished, quality is "acceptable": +15 point(s) just awarded to @carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda)

**Comment by 0crat** [comment]
Quality review completed: +8 point(s) just awarded to @ypshenychka[/z](https://www.0crat.com/u/ypshenychka)

**Comment by 0crat** [comment]
@carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda) this job was assigned to you 5days ago. It will be taken away from you soon, unless you close it, see [§8](http://www.zerocracy.com/policy.html#8). Read [this](http://www.yegor256.com/2014/04/13/no-obligations-principle.html) and [this](http://www.yegor256.com/2014/11/24/principles-of-bug-tracking.html), please.

**Comment by carlosmiranda** [comment]
@0crat status

**Comment by 0crat** [comment]
> @0crat status ([here](https://github.com/zerocracy/farm/issues/1181#issuecomment-399808336))

@carlosmiranda This is what I know about this job in [C3NDPUA8L](https://www.0crat.com/p/C3NDPUA8L), as in [§32](http://www.zerocracy.com/policy.html#32):

  * The job [#1181](https://github.com/zerocracy/farm/issues/1181) is [in scope](http://www.zerocracy.com/policy.html#14) for 6days
  * The role is `REV`
  * The job is assigned to @carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda) for [6days](http://www.zerocracy.com/policy.html#8)
  * There is a monetary reward attached
  * @carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda) will get 25% of their hourly rate for this job
  * Some equity will be [vested](http://www.zerocracy.com/policy.html#37) on completion
  * The job doesn't have any [impediments](http://www.zerocracy.com/policy.html#9)
  * The [budget](http://www.zerocracy.com/policy.html#4) is 15 minutes/points
  * These users are banned and won't be assigned:
    * @krzyk[/z](https://www.0crat.com/u/krzyk): This user reported the ticket
  * Job [footprint](https://www.0crat.com/footprint/C3NDPUA8L?q=%7Bjob%3A%27gh%3Azerocracy%2Ffarm%231181%27%2Ctype%3A%7B%24not%3A%2F%5ENotify%2F%7D%7D) (restricted area)

**Comment by carlosmiranda** [comment]
@g4s8 looks like @0crat is confused about the status here, can you do `@0crat out` on here please?

**Comment by 0crat** [comment]
Are you speaking to me or about me [here](https://github.com/zerocracy/farm/issues/1181#issuecomment-399808683); you must always start your message with my name if you want to address it to me, see [§1](http://www.zerocracy.com/policy.html#1)

**Comment by g4s8** [comment]
@0crat out

**Comment by 0crat** [comment]
@ypshenychka[/z](https://www.0crat.com/u/ypshenychka) please review this job completed by @carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
> @0crat out ([here](https://github.com/zerocracy/farm/issues/1181#issuecomment-399809220))

@g4s8 The job [#1181](https://github.com/zerocracy/farm/issues/1181) is now out of scope

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @g4s8[/z](https://www.0crat.com/u/g4s8)

**Comment by ypshenychka** [comment]
@0crat quality acceptable

**Comment by 0crat** [comment]
Quality review completed: +8 point(s) just awarded to @ypshenychka[/z](https://www.0crat.com/u/ypshenychka)

**Pull request merged**